pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -78,15 +78,24 @@ func (b *AutoscalingGroupModelBuilder) Build(c *fi.ModelBuilderContext) error {
 				SecurityGroups: []*awstasks.SecurityGroup{
 					b.LinkToSecurityGroup(ig.Spec.Role),
 				},
-				IAMInstanceProfile: b.LinkToIAMInstanceProfile(ig),
-				ImageID:            s(ig.Spec.Image),
-				InstanceType:       s(ig.Spec.MachineType),
-
+				IAMInstanceProfile:     b.LinkToIAMInstanceProfile(ig),
+				ImageID:                s(ig.Spec.Image),
+				InstanceType:           s(ig.Spec.MachineType),
 				RootVolumeSize:         i64(int64(volumeSize)),
 				RootVolumeType:         s(volumeType),
 				RootVolumeOptimization: ig.Spec.RootVolumeOptimization,
 			}
 
+			if ig.Spec.SecurityGroup != nil {
```

Comment:
Can you put the SecurityGroups default-case initialization into an else-block of this if statement, as otherwise it's confusing as to what's happening.